src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_io_net_stm32f4_nucleo_wifi.c

```diff
@@ -107,28 +108,38 @@ xi_bsp_io_net_state_t xi_bsp_io_net_read( xi_bsp_socket_t xi_socket,
 
     *out_read_count = 0;
 
-    if ( xi_wifi_state == xi_wifi_state_connected )
+    if ( xi_net_state.wifi_state == xi_wifi_state_connected )
     {
-        if ( xi_wifi_buffer.size == 0 )
+        if ( xi_net_state.head == NULL )
```

Comment:
we usually do this reverse NULL == xi_net_state.head no lower the chance of setting value of the variable accidentally.